x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -189,15 +194,17 @@ public void createResponse(String docId,
                                R response,
                                ActionListener<IndexResponse> listener) throws IOException {
         try {
-            final ReleasableBytesStreamOutput buffer = new ReleasableBytesStreamOutput(0, bigArrays.withCircuitBreaking());
+            final ReleasableBytesStreamOutput buffer = new ReleasableBytesStreamOutputWithLimit(
```

Comment:
Currently we do not create a failure response for this method. This can be an issue for EQL. Should we disable this the limit in this method and integrate it later?

Reply:
I think we don't need to store a failure here, the whole request just fails and returns 400. The same thing will have if there is circuit breaker exception. I think for async search the current implementation is fine.

But you are right I am not sure what is the implication for EQL, and since we did not give them enough time, may be worth not to include this part for EQL

Reply:
@dnhatn  Thanks for the comment. For now, I've created a separate `AsyncTaskIndexService::createResponseForEQL` method for EQL that would create a failure for EQL.  Later we can remove this method, and ask EQL team to double check.

Addressed in d03015c22b4ddf3b573fbef2f78003ab3f8c1603